src/mru_app_list.ts

```diff
@@ -0,0 +1,64 @@
+// @ts-ignore
+const Me = imports.misc.extensionUtils.getCurrentExtension();
+
+import * as cfm from 'config_file_manager';
+import * as launch from "launcher_service";
+
+const MRU_FILE = "mru_apps.json";
+
+export class MruList {
+  entries = new Array<string>();
+
+  constructor() {
+    this.reload();
+  }
+
+  add_recent(option: launch.SearchOption) {
+    if (!("app" in option.id)) {
+      return;
+    }
+
+    const id = option.id.app.filename;
+    const index = this.entries.indexOf(id);
+    if (index >= 0) {
+      this.entries.splice(index, 1);
+    }
+
+    this.entries.push(id);
+
+    const over = this.entries.length - 5;
```

Comment:
Should probably move 5 to a constant variable at the top of the file to make it easier to find and adjust the list length :rofl: